### PR TITLE
Feat: navmap full zoom

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?

Brings possibility to make max zoom in the NavMap, so that full map is visible.
- increased max value for zoom 
- add 2 more zoom level steps
- fixed zoom levels curve evaluation
- fixed clamp for full map zoom

## How to test the changes?

- **Feature**
1. Launch the explorer
2. Open NavMap
3. Play with zoom and map drag. 
4. Verify you can see full map and drag is working properly.

- **Feature Flag**
1. Launch the branch without feature flag
2. Verify you can not do full zoom out
3. Note: it will still have additional 2 zoom steps when you scroll

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 0f585f0</samp>

This pull request enhances the map renderer and UI by improving the camera movement logic, refactoring the zoom controller, adjusting the zoom curve and steps, simplifying the map prefabs, and adding a feature flag for the zoom slider range.

### CODE REVIEW

**Review by RominaMarchetti — APPROVED** [review]
Working as expected when feature flag enable and disable.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapZoomView.cs [hunk 1]
```diff
@@ -18,6 +18,24 @@ public class NavmapZoomView : MonoBehaviour
         internal ZoomInput ZoomOut => zoomOut;
         internal InputAction_Measurable MouseWheelAction => mouseWheelAction;
 
+        private void Awake()
+        {
+            if (DataStore.i.featureFlags.flags.Get().IsInitialized)
```
Comment (davidejensen) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapZoomView.cs:23]: would be better to inject the DataStore from the controller as we do usually with an Initialize method or something like that

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapZoomViewController.cs [hunk 2]
```diff
@@ -23,27 +26,50 @@ public class NavmapZoomViewController : IDisposable, INavmapZoomViewController
...
-            keys[0] = firstKey;
+            CurveClamp01();
+        }
 
-            var lastKey = keys[^1];
-            lastKey.value = 1;
-            keys[^1] = lastKey;
+        public void Dispose()
+        {
+            if (cts != null)
```
Comment (davidejensen) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapZoomViewController.cs:41]: you can use the `.SafeCancelAndDispose();`

**Review by davidejensen — APPROVED** [review]
Approved with 2 minor suggestions

**Review by anicalbano — APPROVED** [review]
✅ Working as expected with and without the FF

https://github.com/decentraland/unity-renderer/assets/139157886/6adf01e9-0e3f-4898-82be-7da6372011f5

**Pull request merged**